R/Rules-methods.R

```diff
@@ -172,7 +171,8 @@ setMethod(
     prob_target <- colMeans(h_in_range(prob_samples, nextBest@target))
 
     # Eligible grid doses after accounting for maximum possible dose and discarding overdoses.
-    is_dose_eligible <- (data@doseGrid <= doselimit) & (prob_overdose < nextBest@max_overdose_prob)
+    is_dose_eligible <- h_next_best_eligible_doses(data@doseGrid, doselimit, data@placebo, levels = TRUE)
+    is_dose_eligible <- is_dose_eligible & (prob_overdose < nextBest@max_overdose_prob)
```

Comment:
maybe better to combine into one step here. otherwise line 175 reads a bit confusing